[Cache] does not work with Redis through unix socket in version 4.2


Symfony version(s) affected: 4.2.2
Description
After upgrade symfony/cache  from v4.1.9 to v4.2.2, get error:

Redis connection failed (php_network_getaddresses: getaddrinfo failed: No address associated with hostname): redis:///var/run/redis/redis.sock/2

How to reproduce
Symfony\Component\Cache\Adapter\RedisAdapter::createConnection('redis:///var/run/redis/redis.sock/2', ['persistent' => 1, 'persistent_id' => 'cache_']);
Possible Solution
Fix this:
 @$redis->{$connect}($hosts[0]['host'], $hosts[0]['port'], $params['timeout'], (string) $params['persistent_id'], $params['retry_interval']);
Must be:
 @$redis->{$connect}($hosts[0]['host'] ?? $hosts[0]['path'], $hosts[0]['port'], $params['timeout'], (string) $params['persistent_id'], $params['retry_interval']);
